### PR TITLE
Include Java version on generating classes jar file to resolve UnsupportedClassVersion

### PR DESCRIPTION
**Pull request by annaibm**
- Include Java version number when generating the classes.jar file as classes_jdk8.jar.
closes: https://github.com/eclipse-openj9/openj9/issues/19582

### CODE REVIEW

**Comment by annaibm** [comment]
Grinder links:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41139/console - JDK8
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41122/ - JDK11
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41123/console - JDK 21,8

**Comment by annaibm** [comment]
Grinder links:
TARGET: extended.system on master branch
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41271/ -windows JDK8
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41270/ - linux JDK8

TARGET: SharedClasses.SCM01.SingleCL
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41273/ - JDK 17
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41274/ - JDK 21

Grinder run On PR branches for issue: [4912](https://github.com/adoptium/aqa-tests/issues/4912) - 
aqa test: PR: [5361](https://github.com/adoptium/aqa-tests/pull/5361) 
TKG: PR: [564](https://github.com/adoptium/TKG/pull/564).
STF: PR: [138](https://github.com/adoptium/STF/pull/138) 

TARGET: SharedClasses.SCM01.SingleCL
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41261/ -JDK 8
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41262/ -JDK11
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41263/ -JDK17
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/41264/ -JDK 21

**Comment by llxia** [comment]
```
03:32:01.370  GEN C:/Users/jenkins/workspace/Grinder/jvmtest/system/systemtest_prereqs/sharedClassesTestData/v1/classes_jdk8.jar does not exist
...
03:52:04.468  STF 17:50:46.067 - +------ Step 2 - Copy sharedClasses jar
03:52:04.468  STF 17:50:46.067 - | Copy a file to another directory
03:52:04.468  STF 17:50:46.067 - |   Source file: C:/Users/jenkins/workspace/Grinder/jvmtest/system/systemtest_prereqs/sharedClassesTestData/v1/classes_jdk8.jar
```

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java [hunk 1]
```diff
@@ -186,16 +186,21 @@ public void setUp(StfCoreExtension test, StfSharedClassesExtension sharedClasses
 		// See whether the test data (20000 classes and jars) already exist.
 		// If not, create them in the first -systemtest-prereqs directory (or the default location if no -systemtest-prereqs were supplied).
 		DirectoryRef sharedClassesDataDir = null;   // This will be initialised when we find it.
+		int javaVersion = test.env().primaryJvm().getJavaVersion();
+		System.out.println("java.version = " + javaVersion);
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java:190]: we can remove this println

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java [hunk 2]
```diff
@@ -91,64 +92,72 @@ public static void main(String[] args)
...
+		generator.go(path, numArg, javaVersion);
+
 	}
-	
-	void go(String path, int numArg) {
+
+	void go(String path, int numArg, int  javaVersion) {
 		log("Creating source files");
-		makeJavas(path, numArg); //Make java source files
+		makeJavas(path, numArg,javaVersion); //Make java source files
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:101]: - need extra space
- is `javaVersion` used in makeJavas()?

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java [hunk 3]
```diff
@@ -91,64 +92,72 @@ public static void main(String[] args)
...
 		log("Compiling java files");
 		compileJavas(path); //Compile Source to .class files
 		log("Source files compiled, creating jar files");
-		makeJars(path, numArg); //Create jar files
+		makeJars(path, numArg,javaVersion); //Create jar files
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:105]: need extra space

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java [hunk 4]
```diff
@@ -433,79 +442,87 @@ private void makeJars(String directory, int numJars)
...
 		{
 			File jarDir = new File(directory + sl + "jars");
+			if (jarDir.exists()) {
+				// Attempt to delete it
+				try {
+					deleteDirectory(jarDir);
+				} catch (IOException e) {
+					throw new RuntimeException("Failed to delete existing directory: " + e.getMessage());
+				}
+			}
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:486]: please delete this and deleteDirectory() below

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java [hunk 5]
```diff
@@ -91,64 +92,72 @@ public static void main(String[] args)
...
-		
+
 		File fileDir = new File(directory);
+		if (fileDir.exists()) {
+			try {
+				deleteDirectory(fileDir);
+			} catch (IOException e) {
+				throw new RuntimeException("Failed to delete existing directory: " + e.getMessage());
+			}
+		}
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:136]: Please delete this block

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java [hunk 6]
```diff
@@ -91,64 +92,72 @@ public static void main(String[] args)
...
 	 * Creates a large number of java files.
-	 * 
+	 *
 	 * @param directory String, the directory to write them into.
 	 * @param count int, the number of tests to generate.
 	 */
-	private void makeJavas(String dir, int count)
+	private void makeJavas(String dir, int count, int javaVersion)
 	{
+		System.out.println("dir:" + dir);
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:125]: We can remove the debug message now

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java [hunk 7]
```diff
@@ -211,7 +216,8 @@ public void setUp(StfCoreExtension test, StfSharedClassesExtension sharedClasses
 						.addProjectToClasspath("openj9.test.sharedClasses")
 						.runClass(JavaGen.class)
 						.addArg(sharedClassesDataDir.getSpec())
-						.addArg("10000"));
+						.addArg("10000")
+						.addArg(String.valueOf(javaVersion)));
```
Comment (llxia) [inline openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java:220]: Do we need to add `javaVersion` as an arg?

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java (hunk 2 above)
Comment (annaibm) [reply openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/test/sc/JavaGen.java:101]: `javaVersion` seems to be not used in makeJavas. Will remove it there.

File: openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java (hunk 7 above)
Comment (annaibm) [reply openj9.test.sharedClasses/src/test.sharedClasses/net/openj9/stf/SharedClasses.java:219]: I am passing the `javaVersion` as an argument to be used by J`avaGen.class` file, or should we set it as environment Variable or as a static field ?

**Comment by annaibm** [comment]
Closing this PR , will raise new PR to address the issue.

**Pull request closed without merging**